rate_limiter/src/lib.rs

```diff
@@ -87,8 +84,7 @@ fn gcd(x: u64, y: u64) -> u64 {
 
 /// TokenBucket provides a lower level interface to rate limiting with a
 /// configurable capacity, refill-rate and initial burst.
-#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
-#[serde(deny_unknown_fields)]
+#[derive(Clone, Debug, Default, PartialEq)]
 pub struct TokenBucket {
```

Comment:
Since you are removing the serde macros, I think it's a good idea to also use `TokenBucketConfig` in the structure instead of having `size`, `one_time_burst` and `refill_time` as separate fields.

Reply:
I tried doing this, but `TokenBucketConfig` is defined in the `vmm` package -- and since this package depends on `rate_limiter`, we can't make `rate_limiter` depend on it to get `TokenBucketConfig` this way. So the only way in which to do this is to move `TokenBucketConfig` to `rate_limiter`. However, this means that `serde` will still have to be included and used in `rate_limiter`, which seems to defeat the point of the initial issue. Since this state is private anyway, wouldn't the cleanest solution be to keep the fields separate as they were initially?

Reply:
We need to do some other refactoring in how we handle structures in Firecracker. I've open an issue to discuss about this as well (#1167) so we can leave it as is right now until we figure out what is the best ownership model for the structures.

Reply:
Ok, sure, thank you!